test/CMakeLists.txt

```diff
@@ -1,226 +1,221 @@
-add_test(GroupByTest GroupByTest)
 target_link_libraries(GroupByTest gtest_main engine ${CMAKE_THREAD_LIBS_INIT})
+gtest_discover_tests(GroupByTest)
 
 add_executable(VocabularyGeneratorTest VocabularyGeneratorTest.cpp)
-add_test(VocabularyGeneratorTest VocabularyGeneratorTest)
 target_link_libraries(VocabularyGeneratorTest gtest_main index ${CMAKE_THREAD_LIBS_INIT})
+gtest_discover_tests(VocabularyGeneratorTest)
 
 add_executable(HasPredicateScanTest HasPredicateScanTest.cpp)
-add_test(HasPredicateScanTest HasPredicateScanTest)
 target_link_libraries(HasPredicateScanTest gtest_main engine ${CMAKE_THREAD_LIBS_INIT})
+gtest_discover_tests(HasPredicateScanTest)
 
 add_executable(MmapVectorTest MmapVectorTest.cpp)
-add_test(MmapVectorTest MmapVectorTest)
 target_link_libraries(MmapVectorTest gtest_main -pthread)
+gtest_discover_tests(MmapVectorTest)
 
 add_executable(BufferedVectorTest BufferedVectorTest.cpp)
-add_test(BufferedVectorTest BufferedVectorTest)
 target_link_libraries(BufferedVectorTest gtest_main -pthread)
+gtest_discover_tests(BufferedVectorTest)
 
 add_executable(UnionTest UnionTest.cpp)
-add_test(UnionTest UnionTest)
 target_link_libraries(UnionTest gtest_main engine ${CMAKE_THREAD_LIBS_INIT})
+gtest_discover_tests(UnionTest)
 
 add_executable(TokenTest TokenTest.cpp TokenTestCtreHelper.cpp)
-add_test(TokenTest TokenTest)
 target_link_libraries(TokenTest parser re2 gtest_main -pthread ${ICU_LIBRARIES})
+gtest_discover_tests(TokenTest)
 
 add_executable(TurtleParserTest TurtleParserTest.cpp)
-add_test(TurtleParserTest TurtleParserTest)
 target_link_libraries(TurtleParserTest parser absl::flat_hash_map re2 gtest_main -pthread)
+gtest_discover_tests(TurtleParserTest)
 
 add_executable(MultiColumnJoinTest MultiColumnJoinTest.cpp)
-add_test(MultiColumnJoinTest MultiColumnJoinTest)
 target_link_libraries(MultiColumnJoinTest gtest_main engine ${CMAKE_THREAD_LIBS_INIT})
+gtest_discover_tests(MultiColumnJoinTest)
 
 add_executable(IdTableTest IdTableTest.cpp)
-add_test(IdTableTest IdTableTest)
 target_link_libraries(IdTableTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
+gtest_discover_tests(IdTableTest)
 
 add_executable(TransitivePathTest TransitivePathTest.cpp)
-add_test(TransitivePathTest TransitivePathTest)
 target_link_libraries(TransitivePathTest engine gtest_main ${CMAKE_THREAD_LIBS_INIT})
+gtest_discover_tests(TransitivePathTest)
 
 add_executable(SparqlLexerTest SparqlLexerTest.cpp)
-add_test(SparqlLexerTest SparqlLexerTest)
 target_link_libraries(SparqlLexerTest parser gtest_main ${CMAKE_THREAD_LIBS_INIT} ${ICU_LIBRARIES})
+gtest_discover_tests(SparqlLexerTest)
 
 set(TEST_COMPILE_TIME_REGEXES OFF CACHE BOOL "Include the tests for the Utf8Regex.")
 if (TEST_COMPILE_TIME_REGEXES)
   add_executable(Utf8RegexTest Utf8RegexTest.cpp)
-  add_test(Utf8RegexTest Utf8RegexTest)
   target_link_libraries(Utf8RegexTest gtest_main ${CMAKE_THREAD_LIBS_INIT})
+  gtest_discover_tests(Utf8RegexTest)
 endif (TEST_COMPILE_TIME_REGEXES)
```

Comment:
@joka921 This test is pretty much empty and unused. What was the rationale here? ctre seems to support UTF-8 when using the correct unicode header, see https://github.com/hanickadot/compile-time-regular-expressions#unicode-support

Can those tests be fixed or can they just be removed since they're unused anyways?